lib/Frontend/PrintingDiagnosticConsumer.cpp

```diff
@@ -945,6 +952,27 @@ void PrintingDiagnosticConsumer::flush(bool includeTrailingBreak) {
 bool PrintingDiagnosticConsumer::finishProcessing() {
   // If there's an in-flight snippet, flush it.
   flush(false);
+
+  // If educational notes are disabled AND a note would have been printed if
+  // they were enabled, print a fake remark letting the user know the additional
+  // information is available. We don't route a real remark through the
+  // DiagnosticEngine because this message should only appear in printed output.
```

Comment:
When exactly is this remark presented to the user? For example, if someone is building from the command line, will they always get this remark when they don't have educational notes enabled but happen to encounter an error that has documentation? I understand the discoverability issue, but I imagine this might annoy people who don't want the see the extra documentation

Reply:
Yeah, your interpretation is correct, the criteria is that edu notes are disabled and a diagnostic with documentation was emitted somewhere.

I've gone back and forth a few times trying to decide if this remark is a good idea or not. FWIW, I stole the idea from Rust, which does this:
<img width="906" alt="Screen Shot 2020-03-25 at 12 39 43 PM" src="https://user-images.githubusercontent.com/1946221/77578685-36195100-6e96-11ea-9314-1b8bc6bf36ad.png">

One option if we drop this message would be to reintroduce it later if we ever introduce some kind of global compiler configuration file, which could have a setting to suppress it. I imagine we'll want to have something of the sort for the language settings introduced by the proposed GSoC project.

Reply:
I'm reverting this change for now, I'll reconsider later if it seems like there's a discoverability problem in practice.